content/gotraining.slide

```diff
@@ -9,10 +9,229 @@ https://github.com/anz-bank/go-training
 
 * Introduction
 
-* Test
 
-# * Go history
 
-# * Why Go at ANZ
+* Agenda (11 - 15 February)
 
-# * Ping server Java Spring Boot vs Go
+- Introduction to Go
+- Go language fundamentals
+- Go standard library
+- Go tool chain
+- Third party Go libraries used at ANZ
+- Project work
+
+Find Slides at [[https://gotraining.appspot.com]] ([[https://github.com/anz-bank/go-slides][source]] on GitHub)
+
+
+* Day 1 - Schedule
+
+*Morning:* Introduction to Go
+
+- Overview and history
+- Demo
+
+*Afternoon:* Hello world and first PR
+
+- Interlude: towards continuous deployment (CD)
+- Go language fundamentals
+- First lab
+
+* History and Overview
+
+* History of Go
+
+.image img/xkcd_compiling.png
+.caption [[https://xkcd.com/303/]]
+
+2007: Robert Griesemer, Rob Pike, and Ken Thompson start working on go at Google
+2008: Russ Cox (go mod) and Ian Taylor (gcc frontend) join
+2009: Go goes Open Source
+2012: Go version 1.0
+
+
+* What is Go?
+
+- A simple language that is easy to learn and read
+- Statically typed, but with a dynamic feel
+- Compiled to native machine code, but has a fast development cycle
+- Language-level concurrency features
+- Expansive "batteries included" standard library
+- Great tools
+- Open source
+
+(Source: [[https://talks.golang.org/2012/zen.slide#2][Go and the Zen of Python]])
+
+* What does Go look like?
+
+Hello world in Go
+
+	package main
+
+	import (
+		"fmt"
+	)
+
+	func main() {
+		fmt.Println("Hello, ಬೆಂಗಳೂರು")
+	}
+
+Execute with
+
+	go run hello.go
+
+* What is NOT Go?
+
+- No type hierarchies or inheritance
+- No exceptions
+- No method overloading
+- No generics
+- No iterators
+- No decorators
+- No named or optional arguments
+- No generators
+- No operator overloading
+- No macros
+
+* Motivation and Demo
+
+* Why Go at ANZ
+
+- simple: easy to learn and read
+- idiomatic
+- tools - gofmt, goimports, go mod, go test, golangci-lint, coverage
+- standard library
+- productive and fun
+- concurrency
+
+* Demo
+- 2 seconds latency Pingserver
+- Java Spring Boot implementation
+- Go implementation
+- Benchmark
+
+* Java Spring Boot Initializr
+
+	── build.gradle
+	├── gradle
+	│   └── wrapper
+	│       ├── gradle-wrapper.jar
+	│       └── gradle-wrapper.properties
+	├── gradlew
+	├── gradlew.bat
+	├── settings.gradle
+	└── src
+	    ├── main
+	    │   ├── java
+	    │   │   └── com
+	    │   │       └── anz
+	    │   │           └── demo
+	    │   │               └── pingserver
+	    │   │                   ├── PingserverApplication.java
+	    │   │                   └── controller
+	    │   │                       └── PingController.java        <--------- THAT'S THE ONE
+	    │   └── resources
+	    │       └── application.properties
+	    └── test
+	          .....
+		                                └── PingserverApplicationTests.java
```

Comment:
I love this

Reply:
The `....` or the tree view? you are up to 1% certificate now.